scripts/globals/dark_ixion.lua

```diff
@@ -396,6 +416,101 @@ xi.darkixion.onMobSpawn = function(mob)
     mob:setAggressive(true)
 end
 
+xi.darkixion.onCriticalHit = function(mob, attacker)
+    local random = math.random(1, 100)
+
+    if
+        (mob:getAnimationSub() == 0 or mob:getAnimationSub() == 3) and
+        (attacker ~= nil and attacker:isInfront(mob)) and
+        random <= 5
+    then
+        mob:setAnimationSub(2)
+        mob:setLocalVar('double', 0)
+        mob:hideHP(false)
+    end
+end
+
+xi.darkixion.onWeaponskillHit = function(mob, attacker, weaponskill)
+    if
+        (mob:getAnimationSub() == 0 or mob:getAnimationSub() == 3) and
+        (attacker ~= nil and attacker:isInfront(mob)) and
+        math.random(1, 100) <= 5
+    then
+        mob:AnimationSub(2)
+        mob:setLocalVar('double', 0)
+        mob:hideHP(false)
+    end
+
+    return 0
+end
+
+xi.darkixion.onMobWeaponSkillPrepare = function(mob, target)
+    -- skill unknown, tp not reset yet
+    -- preserve tp when doing melee swings, due to SetMobSkillAttack
+    local tpPerHit = 64 -- give tp for the auto attack (amount confirmed by not replacing autos with mobskill list)
+    mob:setLocalVar('skill_tp', mob:getTP() + tpPerHit)
+end
+
+xi.darkixion.onMobWeaponSkill = function(target, mob, skill)
+    -- skill chosen, tp already wiped
+    local skillID = skill:getID()
+    if
+        skillID == 2341 or
```

Comment:
Can you give these comments or extract them into local vars that say the name please?